src/injector.ts

```diff
@@ -206,6 +223,9 @@ export class Injector {
             return { tokens: [ typeRef.token ] };
         else if (typeRef.isLiteral())
             return { tokens: [ typeRef.value.constructor ]}
+        else if (typeRef.isGeneric() )
+          // TODO: ask @rezonant for a better way to handle this
+          return { tokens: [ typeRef ]}
```

Comment:
Thinking this is good now. Essentially we support the ref itself as an actual token (and presume our token resolution will unify different instances)